From: DoloresPark <dolorespark@e-vertise.com>
Date: Sat, 23 Aug 2014 13:46:29 -0400
Subject: fix API version mismatch for 'security_compute_relabel()'

diff --git a/android/selinux/stubs.c b/android/selinux/stubs.c
index 7bcbc95..0ad157a 100644
--- a/android/selinux/stubs.c
+++ b/android/selinux/stubs.c
@@ -19,7 +19,8 @@ int get_default_context(const char* user,
 /* Compute a relabeling decision and set *newcon to refer to it.
    Caller must free via freecon.
    Stub not implemented in bionic, but declared in selinux.h */
-#if defined(BIONIC_L) || !defined(__i386__)
+/*#if defined(BIONIC_L) || !defined(__i386__)*/
+#if defined(BIONIC_L)
 int security_compute_relabel(const char *scon UNUSED_PARAM,
 	const char *tcon,
 	security_class_t tclass UNUSED_PARAM,
