From: DoloresPark <dolorespark@e-vertise.com>
Date: Mon, 25 Aug 2014 16:07:31 -0400
Subject: AOSP's libext2_uuid has a '_static' suffix

diff --git a/Android.mk b/Android.mk
index b8e42c7..c4a0ffe 100644
--- a/Android.mk
+++ b/Android.mk
@@ -124,7 +124,7 @@ LOCAL_CFLAGS := $(common_cflags) -DMINIVOLD
 LOCAL_STATIC_LIBRARIES := libcm_minivold
 LOCAL_STATIC_LIBRARIES += libc libstdc++
 LOCAL_STATIC_LIBRARIES += $(common_libraries) $(common_static_libraries)
-LOCAL_STATIC_LIBRARIES += libcrypto_static libext2_uuid libcm_vold
+LOCAL_STATIC_LIBRARIES += libcrypto_static libext2_uuid_static libcm_vold
 LOCAL_FORCE_STATIC_EXECUTABLE := true
 LOCAL_MODULE_CLASS := RECOVERY_EXECUTABLES
 LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/sbin
