From: DoloresPark <dolorespark@e-vertise.com>
Date: Thu, 21 Aug 2014 21:47:34 -0400
Subject: send mkfs.f2fs to /system/bin rather than /sbin

diff --git a/Android.mk b/Android.mk
index bee1071..dbe6baf 100644
--- a/Android.mk
+++ b/Android.mk
@@ -68,7 +68,9 @@ LOCAL_FORCE_STATIC_EXECUTABLE := true
 # Directly generating into the recovery/root/sbin gets clobbered
 # when the recovery image is being made.
 # LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/sbin
-LOCAL_MODULE_PATH := $(TARGET_ROOT_OUT_SBIN)
+#
+# send it to /system/bin - recovery uses busybox
+#LOCAL_MODULE_PATH := $(TARGET_ROOT_OUT_SBIN)
 
 LOCAL_SRC_FILES := \
 	lib/libf2fs_io.c \
