From: DoloresPark <dolorespark@e-vertise.com>
Date: Fri, 22 Aug 2014 11:06:36 -0400
Subject: releasetools: add f2fs as a valid OTA formatting option

diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index a3217dd..f30b3ec 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -946,7 +946,8 @@ def ComputeDifferences(diffs):
 
 # map recovery.fstab's fs_types to mount/format "partition types"
 PARTITION_TYPES = { "yaffs2": "MTD", "mtd": "MTD",
-                    "ext4": "EMMC", "emmc": "EMMC" }
+                    "ext4": "EMMC", "emmc": "EMMC",
+                    "f2fs": "EMMC", "emmc": "EMMC" }
 
 def GetTypeAndDevice(mount_point, info):
   fstab = info["fstab"]
